src/Confluent.Kafka/Producer.cs

```diff
@@ -616,6 +617,24 @@ public int Flush(int millisecondsTimeout)
             => kafkaHandle.Flush(millisecondsTimeout);
 
         /// <summary>
+        ///     Wait until all outstanding produce requests and delievery report
+        ///     callbacks are completed. Refer to <see cref="Flush(int)" /> for
+        ///     more information.
+        /// 
+        ///     [UNSTABLE-API] - the semantics and/or type of the return value is
+        ///     subject to change.
+        /// </summary>
+        /// <param name="timeout">
+        ///     The maximum length of time to block.
```

Comment:
Same remark (avoid TimeStamp.Timeout which may block kafka thread or something like that)
The maximum time to block (no length of)